SIGSEGV when connecting to a hypervisor over qemu+ssh

## Description
The provider crushes while trying to connect to a hypervisor over ssh (go implementation).

## Configuration
```hcl
terraform {
  required_providers {
    libvirt = {
      source  = "dmacvicar/libvirt"
      version = "0.9.0"
    }
  }
}

provider "libvirt" {
  uri = "qemu+ssh://root@<redacted hostname>/system"
}

resource "libvirt_volume" "alpine_base" {
  name   = "alpine-3.22-base.qcow2"
  pool   = "default"
  format = "qcow2"

  create = {
    content = {
      url = "https://dl-cdn.alpinelinux.org/alpine/v3.22/releases/cloud/generic_alpine-3.22.2-x86_64-bios-cloudinit-r0.qcow2"
    }
  }
}
```

## `terraform plan` output

<details>

<summary>terraform plan output</summary>

```
Planning failed. Terraform encountered an error while generating this plan.

╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the plugin6.(*GRPCProvider).ConfigureProvider call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-libvirt_v0.9.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x7da5e4]

goroutine 50 [running]:
github.com/digitalocean/go-libvirt/socket/dialers.WithSystemSSHDefaults(0xc00003c490?)
	github.com/digitalocean/go-libvirt@v0.0.0-20250923171224-1d0cf4034554/socket/dialers/gossh.go:123 +0x24
github.com/dmacvicar/terraform-provider-libvirt/v2/internal/libvirt/dialers.newGoSSHDialer(0xc0000f79e0)
	github.com/dmacvicar/terraform-provider-libvirt/v2/internal/libvirt/dialers/factory.go:106 +0x68
github.com/dmacvicar/terraform-provider-libvirt/v2/internal/libvirt/dialers.NewDialerFromURI({0xc00003c480?, 0xc0005a0150?})
	github.com/dmacvicar/terraform-provider-libvirt/v2/internal/libvirt/dialers/factory.go:54 +0x15a
github.com/dmacvicar/terraform-provider-libvirt/v2/internal/libvirt.NewClient({0x1105708, 0xc0005a0150}, {0xc00003c480, 0x34})
	github.com/dmacvicar/terraform-provider-libvirt/v2/internal/libvirt/client.go:28 +0x17f
github.com/dmacvicar/terraform-provider-libvirt/v2/internal/provider.(*LibvirtProvider).Configure(0xc0005a0540?, {0x1105708, 0xc0005a0150}, {{0xc0004ac392, 0x6}, {{{0x110b398, 0xc0005a05d0}, {0xe2bb20, 0xc0005a0540}}, {0x110e900, ...}}, ...}, ...)
	github.com/dmacvicar/terraform-provider-libvirt/v2/internal/provider/provider.go:95 +0x2d9
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ConfigureProvider(0xc00024a308, {0x1105708, 0xc0005a0150}, 0xc0003a0140, 0xc000232000)
	github.com/hashicorp/terraform-plugin-framework@v1.16.1/internal/fwserver/server_configureprovider.go:18 +0x131
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ConfigureProvider(0xc00024a308, {0x1105708?, 0xc0005a0030?}, 0xc000630020)
	github.com/hashicorp/terraform-plugin-framework@v1.16.1/internal/proto6server/server_configureprovider.go:39 +0x29f
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ConfigureProvider(0xc000248460, {0x1105708?, 0xc0004885d0?}, 0xc0004a0280)
	github.com/hashicorp/terraform-plugin-go@v0.29.0/tfprotov6/tf6server/server.go:607 +0x33e
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ConfigureProvider_Handler({0xf6c9e0, 0xc000248460}, {0x1105708, 0xc0004885d0}, 0xc000480280, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.29.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:735 +0x1a6
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00021e200, {0x1105708, 0xc0005a1110}, 0xc0003955c0, 0xc00031d710, 0x17ca940, 0x0)
	google.golang.org/grpc@v1.75.1/server.go:1431 +0xfe8
google.golang.org/grpc.(*Server).handleStream(0xc00021e200, {0x1106658, 0xc00050a000}, 0xc0003955c0)
	google.golang.org/grpc@v1.75.1/server.go:1842 +0xb95
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.75.1/server.go:1061 +0x7f
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 20
	google.golang.org/grpc@v1.75.1/server.go:1072 +0x11d

Error: The terraform-provider-libvirt_v0.9.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

</details>

## Cause
https://github.com/dmacvicar/terraform-provider-libvirt/blob/v0.9.0/internal/libvirt/dialers/factory.go#L106
`dialers.WithSystemSSHDefaults` needs a non-nil `*user.User` pointer
https://github.com/digitalocean/go-libvirt/blob/main/socket/dialers/gossh.go#L123

--- a/internal/libvirt/dialers/factory.go
+++ b/internal/libvirt/dialers/factory.go
@@ -101,9 +101,14 @@ func newGoSSHDialer(parsedURI *url.URL) (Dialer, error) {
 		return nil, fmt.Errorf("SSH transport requires a hostname")
 	}
 
+	currUser, err := user.Current()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get current user: %w", err)
+	}
+
 	query := parsedURI.Query()
 	opts := []dialers.SSHOption{
-		dialers.WithSystemSSHDefaults(nil), // Use system defaults
+		dialers.WithSystemSSHDefaults(currUser), // Use system defaults
 	}
 
 	// Port

